### PR TITLE
Products

### PR DESCRIPTION
**Pull request by nikodem-bilczewski**
Main changes:
- change API URL and getProducts function
- change useProducts hook
- created ProductDetail page
- created getProduct function
- added products filtering
- fixed pagination

### CODE REVIEW

File: src/components/molecules/AuthForm/useAuthForm.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { useContext } from 'react'
 import { SubmitHandler, useForm } from 'react-hook-form'
 import { logInFn, signUpFn } from 'services/auth'
 
-import AuthContext from 'components/shared/AuthContext'
+import { AuthContext } from 'components/shared/AuthContext'
```
Comment (DanielKlys7) [inline src/components/molecules/AuthForm/useAuthForm.ts]: I do wonder if `AuthContext` truly is a component, also if I haven't mention it yet: https://www.linkedin.com/pulse/how-properly-use-react-context-typescript-digital-colliers/?trk=organization_guest_main-feed-card_feed-article-content

File: src/components/molecules/Filters/Filters.types.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { ChangeEvent } from 'react'
+
+export type FiltersProps = {
+  selectedCategory: string
+  handleCategoryChange: (event: ChangeEvent<HTMLInputElement>) => void
```
Comment (DanielKlys7) [inline src/components/molecules/Filters/Filters.types.ts:5]: Glad you menaged it this way, well done!

File: src/components/molecules/Filters/useFilters.ts
Comment (DanielKlys7) [inline src/components/molecules/Filters/useFilters.ts:1]: That's a proper way of using hooks, awesome!

File: src/components/molecules/PaginationButtons/PaginationButtons.types.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export type PaginationButtonProps = {
+  forcePage: number
```
Comment (DanielKlys7) [inline src/components/molecules/PaginationButtons/PaginationButtons.types.ts:2]: Why is it called `forcePage`? For me it's not descriptive and names should be

File: src/components/molecules/PaginationButtons/index.tsx [hunk 4]
```diff
@@ -2,10 +2,15 @@ import ReactPaginate from 'react-paginate'
...
-const PaginationButtons = ({ pageCount, onChange }: PaginationButtonProps) => (
+const PaginationButtons = ({
+  pageCount,
+  onChange,
+  forcePage,
+}: PaginationButtonProps) => (
   <ReactPaginate
     pageCount={pageCount}
     pageRangeDisplayed={3}
+    forcePage={forcePage - 1}
```
Comment (DanielKlys7) [inline src/components/molecules/PaginationButtons/index.tsx:13]: Is it just a parameter to handle paging on our own?

File: src/components/organisms/Product/index.tsx [hunk 5]
```diff
@@ -1,15 +1,24 @@
...
-    <img
-      src={image}
-      className='mb-4 aspect-[1/1] h-full w-full rounded-md object-fill object-center'
-    />
-    <h3 className='text-3xl'>{name}</h3>
-    <p className='mb-1 text-2xl font-light'>
+    <Link
+      className='mb-10'
+      to={`/products/${id}`}
+      onClick={() => window.scrollTo({ top: 0, behavior: 'smooth' })}
```
Comment (DanielKlys7) [inline src/components/organisms/Product/index.tsx:12]: just a quick note: window.scrollTo doesn't work on every device, you can keep it this way, but keep that information in mind

File: src/components/organisms/ProductsList/Products.types.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    createdAt: string
+    currency: string
+    description: string
+    img: {
+      data: {
+        attributes: {
+          url: string
+        }
+      }
+    }
```
Comment (DanielKlys7) [inline src/components/organisms/ProductsList/Products.types.ts]: Should be flattened if possible

File: src/components/pages/ProductDetails/index.tsx [hunk 7]
```diff
@@ -1,38 +1,67 @@
...
+  }
+
+  return (
+    <div className='flex flex-col justify-around gap-10 my-10 w-4/5 xl:w-3/5 mx-auto'>
+      <div className='mx-auto md:my-26 lg:my-28 flex w-full flex-col gap-7 rounded-lg bg-stone-200 p-7 shadow-xl  lg:flex-row'>
+        <img
+          className='aspect-[1/1] h-full w-full rounded-md object-fill object-center md:mb-0 md:max-w-xl '
+          src={
+            import.meta.env.VITE_BASE_URL +
+            product?.data.attributes.img.data.attributes.url
```
Comment (DanielKlys7) [inline src/components/pages/ProductDetails/index.tsx]: product could be destructized somewhere before to increase readability

File: src/config/axios/index.ts [hunk 8]
```diff
@@ -1,5 +1,9 @@
 import axios from 'axios'
 
+// export const productsAPI = axios.create({
+//   baseURL: import.meta.env.VITE_BASE_URL,
+// })
+
```
Comment (DanielKlys7) [inline src/config/axios/index.ts]: Why is it for?

File: src/components/molecules/PaginationButtons/PaginationButtons.types.ts (hunk 3 above)
Comment (nikodem-bilczewski) [reply src/components/molecules/PaginationButtons/PaginationButtons.types.ts:2]: forcePage is built in attribute in ReactPaginate component, for me it also is not descriptive, but it's the only way I guess

File: src/components/molecules/PaginationButtons/index.tsx (hunk 4 above)
Comment (nikodem-bilczewski) [reply src/components/molecules/PaginationButtons/index.tsx:13]: Kind of- it helped me with keeping proper pagination page when I go back from ProductDetails page to ProductList. If I didn't use it- every time I went back I was on 1st page, even if product I clicked on was on page 4.

File: src/components/molecules/AuthForm/useAuthForm.ts [hunk 9]
```diff
@@ -3,8 +3,7 @@ import { useMutation } from '@tanstack/react-query'
 import { useContext } from 'react'
 import { SubmitHandler, useForm } from 'react-hook-form'
 import { logInFn, signUpFn } from 'services/auth'
-
-import { AuthContext } from 'components/shared/AuthContext'
+import { AuthContext } from 'shared/AuthContext'
```
Comment (DanielKlys7) [inline src/components/molecules/AuthForm/useAuthForm.ts]: I believe that it should go to `contexts` directory

**Pull request merged**